xc/device.ml

```diff
@@ -1866,6 +1847,32 @@ module Backend = struct
 
       let with_dirty_log domid ~f = f()
 
+      let cmdline_of_info ~xs ~dm info restore domid =
+        let common = Dm_Common.cmdline_of_info ~xs ~dm info restore domid in
+        (* Sort the VIF devices by devid *)
+        let nics = List.stable_sort (fun (_,_,a) (_,_,b) -> compare a b) info.Dm_Common.nics in
+        if List.length nics > Dm_Common.max_emulated_nics then debug "Limiting the number of emulated NICs to %d" Dm_Common.max_emulated_nics;
+        (* Take the first 'max_emulated_nics' elements from the list. *)
+        let rec take n xs = match n, xs with
+          | 0, _ -> []
+          | _, [] -> [] (* return a short list *)
+          | n, x :: xs -> x :: (take (n - 1) xs) in
+        let nics = take Dm_Common.max_emulated_nics nics in
+
+        (* qemu need a different id for every vlan, or things get very bad *)
+        let nics' =
+          if List.length nics > 0 then
+            let vlan_id = ref 0 in
```

Comment:
This can be implemented without references using `Stdext.Listext.List.mapi` which provides an index for each map element. 
```
Stdext.Listext.List.mapi (fun vlan_id (mac, bridge, devid) ->
...) nics
```

Reply:
As mentioned on slack, `mapi` is in the stdlib now, so better use `List.mapi` instad